webrender/src/clip_scroll_node.rs

```diff
@@ -410,22 +407,16 @@ impl ClipScrollNode {
         let z0 = -10000.0;
         let z1 =  10000.0;
 
-        let p0 = inv.transform_point4d(&WorldPoint4D::new(cursor.x, cursor.y, z0, 1.0));
-        let p0 = Point3D::new(p0.x / p0.w,
-                              p0.y / p0.w,
-                              p0.z / p0.w);
-        let p1 = inv.transform_point4d(&WorldPoint4D::new(cursor.x, cursor.y, z1, 1.0));
-        let p1 = Point3D::new(p1.x / p1.w,
-                              p1.y / p1.w,
-                              p1.z / p1.w);
+        let p0 = inv.transform_point3d(&WorldPoint3D::new(cursor.x, cursor.y, z0));
```

Comment:
nice transform call :)
For the vector conversion from 2d to 3d, I'm actually a big fan of `cgmath` approach: https://docs.rs/cgmath/0.14.1/cgmath/struct.Vector2.html#method.extend

You call `Vector2::extend(self, z: f32) -> Vector3`. It's more explicit then current euclid's `to_3d` and allows cases like this ^ to look somewhat nicer